app/views/govuk_publishing_components/components/_intervention.html.erb

```diff
@@ -45,6 +49,8 @@
     suggestion_link_text = intervention_helper.accessible_text
   end
 
+  data_attributes[:ga4_intervention_banner] = "" if ga4_tracking
```

Comment:
@andysellick Good point. I have just checked and you're right. It uses a cookie to hide the element, so I can refrain from sending the pageview value if the cookie exists, and the cookie value contains the intervention campaign name. Thanks :+1:

Reply:
Was this change added? I can't find it.

Reply:
Whoops @andysellick - I think writing that comment made me thing that I had "responded" to this, even though I didn't write the code for it 🤦‍♂️.

I've made this change now if you'd like to review it. Thanks

Reply:
Thanks. Your solution looks complicated for something that only has to return either `true` or `undefined`. Is this because we don't know if the banner code has run at this point (and could be visible, or visible but about to be hidden)? Otherwise could you do something like 

```
if (document.querySelector('[data-ga4-intervention-banner]' && !document.querySelector('[data-ga4-intervention-banner][hidden]')
```

Reply:
Thanks @andysellick - good solution. Unfortunately at the point where our Pageview tracker sends it's data, it can't see the presence of `hidden` on the banner. `document.querySelector('[data-ga4-intervention-banner][hidden]')` was returning `null`. Also, trying this:

```Javascript
document.querySelector('[data-ga4-intervention-banner]').hasAttribute('hidden')
```
returns `false`. But if I wrap that check in a setTimeout for 5 seconds, it will then return `true`. So  after those 5 seconds it looks like the intervention code has run. It's probably as you correctly mentioned, the order of the JS running. Load analytics:  `<%= javascript_include_tag "govuk_publishing_components/load-analytics" %>` runs first in `_layout_for_public.html.erb` at line 71, whereas application.js: `<%= javascript_include_tag 'application' %>` runs at the end on line 170.

Should we stick with the cookie solution? 

Just to let you know, I'm going to need to repeat this fix on the cookie banner as well, as I realised that the cookie banner has the exact same issue.

Thanks